src/providers/google/youtube-post.ts

```diff
@@ -0,0 +1,184 @@
+    }
+
+    protected setNextPosition(
+        syncPosition: SyncHandlerPosition,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$ActivityListResponse>
+    ): SyncHandlerPosition {
+        if (!syncPosition.futureBreakId && serverResponse.data.items.length) {
+            syncPosition.futureBreakId = `${this.connection.profile.id}-${serverResponse.data.items[0].id}`;
+        }
+
+        if (_.has(serverResponse, "data.nextPageToken")) {
+            // Have more results, so set the next page ready for the next request
+            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        } else {
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return syncPosition;
+    }
+
+    protected async buildResults(
+        youtube: youtube_v3.Youtube,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$ActivityListResponse>,
+        breakId: string,
+        breakTimestamp?: string
+    ): Promise<SchemaPost[]> {
+        const results: SchemaPost[] = [];
+
+        const activities = serverResponse.data.items;
+        // filter post(upload, comment, bulletin)
+        const posts = activities.filter(activity => [SchemaYoutubeActivityType.UPLOAD, SchemaYoutubeActivityType.COMMENT].includes(activity.snippet.type as SchemaYoutubeActivityType))
+        for (const post of posts) {
+            const postId = `${this.connection.profile.id}-${post.id}`;
+            console.log(post)
+
+            if (postId == breakId) {
+                break;
+            }
+
+            const snippet = post.snippet;
+            const insertedAt = snippet.publishedAt || "Unknown";
+
+            if (breakTimestamp && insertedAt < breakTimestamp) {
+                break;
+            }
+
+            const title = snippet.title || "No title";
+            const description = snippet.description || "No description";
+            const contentDetails = post.contentDetails;
+
+            const activityType = snippet.type;
+            const iconUri = snippet.thumbnails.default.url;
+            // extract activity URI
+            let activityUri = "";
+            switch (activityType) {
+                case SchemaYoutubeActivityType.UPLOAD:
+                    var videoId = contentDetails.upload.videoId;
+                    activityUri = 'https://www.youtube.com/watch?v=' + videoId;
+                    break;
+                case SchemaYoutubeActivityType.COMMENT:
+                    var videoId = contentDetails.comment.resourceId.videoId;
+                    activityUri = 'https://www.youtube.com/watch?v=' + videoId;
+                    break;
+                default:
+                    activityUri = 'Unknown activity type';
+                    break;
+            }
+
+
+            results.push({
+                _id: `youtube-${postId}`,
+                name: title,
+                icon: iconUri,
+                uri: activityUri,
+                type: activityType,
```

Comment:
Type must match the `type` value from schema (https://common.schemas.verida.io/social/post/v0.1.0/schema.json).

ie: `video`

Reply:
> Type must match the `type` value from schema (https://common.schemas.verida.io/social/post/v0.1.0/schema.json).
> 
> ie: `video`

Defined `postType`